packages/app-builder/src/services/data/data-model.tsx

```diff
@@ -0,0 +1,8 @@
+import { type TableModel } from '@app-builder/models';
+import { createSimpleContext } from '@app-builder/utils/create-context';
+
+const DataModelContext = createSimpleContext<TableModel[]>('DataModelContext');
+
+export const DataModelContextProvider = DataModelContext.Provider;
+
+export const useDataModel = DataModelContext.useValue;
```

Comment:
New context to mutualize datamodel fetching for both new pages (it's the same pattern we already have for `currentScnearioIteration`)